### PR TITLE
resolves #4 : QA Pipeline

### LINKED ISSUE #4: Write a Github Action that runs the static analysis tests as described in the https://developer.adobe.com/commerce/testing/guide/static/
We should do this for two separate things:

These actions should be added here: https://github.com/graycoreio/github-actions-magento2

1. [The current version in v2.4-develop ](https://github.com/mage-os/mageos-magento2)
2. [The v2.4.5-p1 release](https://github.com/magento/magento2/releases/tag/2.4.5-p1)

I would expect to see a failure (or a success in Github) that looks like:

Image

### PR DESCRIPTION
**Pull request by DavidLambauer**
This PR introduces a QA Workflow that uses a Matrix to spin up the same jobs with different parameters. Right now, the matrix is set to `supported-versions`, equivalent to Adobe's support lifecycle. I assume the matrix will change, but to me, it makes sense to have the option to test a PR against different variations of dependencies.

For now, the only test that runs is PHPMD. I used the Magento Ruleset. [A complete run of PHPMD for all Magento Extensions takes around an hour](https://github.com/DavidLambauer/mageos-magento2/actions/runs/3517564983/jobs/5895483151), which is way too long. I only run PHPMD on the changed files within a PR to speed up the feedback loop. This works great for `stupid` tools like PHPMD and PHPCS, but it might not be possible for PHPStan, Copy/Paste Detector, or similar tools.

As the current full scan of PHPMD on Magento throws a lot of errors, I assume I messed up some parameters, or Magento works with a baseline. I haven't found one, so I assume this potential baseline is internal. We should discuss that baseline topic in a tech meeting.

The command to run PHPMD: `php vendor/bin/phpmd app/code github dev/tests/static/testsuite/Magento/Test/Php/_files/phpmd/ruleset.xml` 

 fixes mage-os/mageos-magento2#4

### CODE REVIEW

File: app/code/Magento/AdminAdobeIms/App/Action/Plugin/Authentication.php [hunk 1]
```diff
@@ -56,4 +56,9 @@ public function __construct(
 
         $this->_openActions[] = ImsCallback::ACTION_NAME;
     }
+
+    private function unusedMethod()
```
Comment (Vinai) [inline app/code/Magento/AdminAdobeIms/App/Action/Plugin/Authentication.php:60]: What is the purpose of this method? Left over debug code?

Comment (DavidLambauer) [reply app/code/Magento/AdminAdobeIms/App/Action/Plugin/Authentication.php:60]: Can be seen so, yes. I want to execute PHPMD only on the changed files within a PR to speed up the process. To test this behavior, I need a PHPMD error :). Can be removed as soon as I've seen the result and the nightly builds run again.

**Comment by DavidLambauer** [comment]
I wanted to switch to another matrix, which resulted in an issue. Until then, too many actions are triggered. 

Reference: https://github.com/graycoreio/github-actions-magento2/issues/103

**Comment by DavidLambauer** [comment]
That finally seem to work as expected for PHPMD.

CleanShot 2022-12-04 at 16 40 03

**Comment by Vinai** [comment]
@DavidLambauer is this PR still a work-in-progress?

**Comment by DavidLambauer** [comment]
Yeh, I'll work on it. As we now have the GitHub Actions Repo, I'll transfer everything to there and push an action for PHPMD

**Comment by DavidLambauer** [comment]
Will move this to the github actions repo

**Pull request closed without merging**